### ISSUE TITLE
No response (success nor error) with HEIC format.

### ISSUE THREAD

**Issue by scott-schibli** [onurzorluer/react-image-file-resizer#59]
Hello, I am running into problems with compressing a .heic image. It seems that all other image formats work perfectly, but when I use a .heic image, I do not get a response - neither a success or error in my catch. 

I guess my question is, I don't seem to find anywhere how to specifically reject the returned promise from the 'resizeFile' method from the Resizer.imageFileResizer() method, how can I do that? Does this package even support heic (I cant find anything online about .heic files relating to this package at all -- curious).

This is my resizeFile method: 

const resizeFile = (file: File) => {
    return new Promise((resolve) => {
        Resizer.imageFileResizer(
            file,
            1080,
            1080,
            'JPEG',
            80,
            0,
            (uri) => {
                resolve(uri);
            },
            'base64',
        );
    });
};

And this is my try catch which I am calling it from:

files.forEach(async (file, i) => {
            try {
                const image = await resizeFile(file);
                const newFile = dataURIToBlob(image);
                formData.append(file.name, newFile);
            } catch (err) {
                console.log(err);
            }
});

**Expected behavior**
I expect a response, either an error or success from passing any type of image. 

**Desktop (please complete the following information):**
 - OS: [e.g. iOS]: Catalina 10.15.5
 - Browser [e.g. chrome, safari]: Chrome 
 - Version [e.g. 22]: 91.0.4472.114

**Comment by boythan** [comment]
I have same problem.

**Comment by mbyrne00** [comment]
Hey @onurzorluer - any chance of incorporating this suggested fix? Would be nice to support heic in this great simple-to-use lib.

**Comment by mbyrne00** [comment]
This has become less urgent for me, as I forgot that when you don't explicitly accept `heic` files with your file input and you do accept `jpg`, then iOS converts them first.

**Comment by jovana** [comment]
I have also a workaround, which is very easy to use. Using the heic2any script: https://www.npmjs.com/package/heic2any

Before running the resizer check the image type and convert:
```
if (file.type === 'image/heic') {
        // file conversion takes a bit.
        file = await heic2any({
            blob: file,
            quality: 0.5
        });
    }
```

**Comment by Andriy-Kulak** [comment]
> This has become less urgent for me, as I forgot that when you don't explicitly accept `heic` files with your file input and you do accept `jpg`, then iOS converts them first.

Can you explain what you mean by this? I don't get it. iOS on phones converts HEIC to JPEG on the fly? that doesn't make sense

**Comment by Andriy-Kulak** [comment]
> I have also a workaround, which is very easy to use. Using the heic2any script: https://www.npmjs.com/package/heic2any
> 
> Before running the resizer check the image type and convert:
> 
> ```
> if (file.type === 'image/heic') {
>         // file conversion takes a bit.
>         file = await heic2any({
>             blob: file,
>             quality: 0.5
>         });
>     }
> ```

When I try to resize the result of this I get the same error as the original Post on here. Do you do anything specific afterwards?

**Comment by mbyrne00** [comment]
> > This has become less urgent for me, as I forgot that when you don't explicitly accept `heic` files with your file input and you do accept `jpg`, then iOS converts them first.
> 
> Can you explain what you mean by this? I don't get it. iOS on phones converts HEIC to JPEG on the fly? that doesn't make sense

Asking for help and telling someone their comment "doesn't make sense" isn't always the best strategy ;-).

Anyway, I don't have the code at hand but if you search around you will see there are various scenarios where iOS devices will automatically convert HEIC to JPG if it's for image requests that do not accept HEIC. For example comments like the one below. I can't remember in the context of this component, but I had a similar experience from memory. For my use cases, this was enough. There may still be valid reasons others want to do it explicitly, however.

>
> Tried a standard input tag with the proper accept attribute
>  ` <input type="file" accept="image/jpeg,image/png" />`
> Selected a HEIC file from Photos in Safari, the selected image was automatically converted to JPEG.
>

**Comment by Andriy-Kulak** [comment]
> > > This has become less urgent for me, as I forgot that when you don't explicitly accept `heic` files with your file input and you do accept `jpg`, then iOS converts them first.
> > 
> > 
> > Can you explain what you mean by this? I don't get it. iOS on phones converts HEIC to JPEG on the fly? that doesn't make sense
> 
> Asking for help and telling someone their comment "doesn't make sense" isn't always the best strategy ;-).
> 
> Anyway, I don't have the code at hand but if you search around you will see there are various scenarios where iOS devices will automatically convert HEIC to JPG if it's for image requests that do not accept HEIC. For example comments like the one below. I can't remember in the context of this component, but I had a similar experience from memory. For my use cases, this was enough. There may still be valid reasons others want to do it explicitly, however.
> 
> > Tried a standard input tag with the proper accept attribute
> > ` <input type="file" accept="image/jpeg,image/png" />`
> > Selected a HEIC file from Photos in Safari, the selected image was automatically converted to JPEG.

You're right. sorry wasn't in the best headspace yesterday. thanks for calling my attitude out and quick response/tip to help me resolve this. The solution that seems to work is specifically defining accept image file types to png jpeg, example below.

```ts
  const {
    ...
    acceptedFiles,
  } = useDropzone({
    accept: {
      "image/jpeg": [],
      "image/png": [],
    }
  });
```
`<input type="file" accept="image/jpeg,image/png" />` as you suggested will probably work just as well.

just tested and seems to work on mobile iphones well. if the desire is to add ability for people to upload HEIC images from desktop, then something like `heic2any` will likely be needed although it's not a priority for me either.

Cheers!
